src/Reference/HasOneSql.php

```diff
@@ -86,31 +89,31 @@ function (Model $m) use ($their_field) {
      * addFields(['from', 'to'], ['type' => 'date']);
      *
      * @param array $fields
-     * @param array $defaults
+     * @param array $groupDefaults
      *
      * @return $this
      */
-    public function addFields($fields = [], $defaults = [])
+    public function addFields($fields = [], $groupDefaults = [])
     {
-        foreach ($fields as $field => $alias) {
-            if (is_array($alias)) {
-                $d = array_merge($defaults, $alias);
-                if (!isset($alias[0])) {
+        foreach ($fields as $fieldName => $defaults) {
+            if (is_array($defaults)) {
+                $field = array_merge($groupDefaults, $defaults);
+                if (!isset($defaults[0])) {
                     throw (new Exception('Incorrect definition for addFields. Field name must be specified'))
-                        ->addMoreInfo('field', $field)
-                        ->addMoreInfo('alias', $alias);
+                        ->addMoreInfo('fieldName', $fieldName)
+                        ->addMoreInfo('alias', $defaults);
                 }
-                $alias = $alias[0];
+                $defaults = $defaults[0];
             } else {
-                $d = $defaults;
+                $field = $groupDefaults;
```

Comment:
to be removed, right? (and not if above)

Reply:
Simplified